### ISSUE TITLE
Proposal - unified API for ordinations?

### ISSUE THREAD

**Issue by kescobo** [JuliaStats/MultivariateStats.jl#95]
I'm not sure if this is possible / correct, but it would be nice if there were a unified API for ordinations like MDS and PCA to extract the scaled axes and other components. This arose in https://github.com/JuliaPlots/StatsPlots.jl/pull/229 and https://github.com/JuliaPlots/StatsPlots.jl/pull/230. 

Having something like `AbstractOrdination` would make it possible to have a single plot recipe if there were functions for accessing similar components, esp. the coordinates along each axis and the % variation explained.

**Comment by wildart** [comment]
So, you want a supertype for MDS and PCA, plus some functional interface for accessing eigenvalues?

**Comment by kescobo** [comment]
Yes, that's what I was thinking - not sure if things like LDA or any others could be included here (I'm not really familiar with the others, I've just seen some of them plotted in similar ways).

For MDS and PCA at least, the components are there. `projection()` is already shared and can be used to extract the actual values to plot, if they shared a function to get the eigenvalues, that would be sufficient (though having a supertype would be cleaner I think).
